src/pages/workspaces/workspace/analysis/modals/compute-modal-components.test.ts

```diff
@@ -0,0 +1,155 @@
+} from 'src/pages/workspaces/workspace/analysis/modals/compute-modal-components'
+import {
+  defaultAutopauseThreshold,
+  defaultComputeRegion,
+  defaultComputeZone,
+  defaultDataprocMachineType,
+  defaultDataprocMasterDiskSize,
+  defaultDataprocWorkerDiskSize,
+  defaultGcePersistentDiskSize,
+  defaultGpuType,
+  defaultNumDataprocPreemptibleWorkers,
+  defaultNumDataprocWorkers,
+  defaultNumGpus,
+  defaultPersistentDiskType,
+  getDefaultMachineType,
+} from 'src/pages/workspaces/workspace/analysis/runtime-utils'
+import { toolLabels } from 'src/pages/workspaces/workspace/analysis/tool-utils'
+
+
+const defaultIComputeConfig = {
+  selectedPersistentDiskSize: defaultGcePersistentDiskSize,
+  selectedPersistentDiskType: defaultPersistentDiskType,
+  masterMachineType: getDefaultMachineType(false, toolLabels.RStudio),
+  masterDiskSize: defaultDataprocMasterDiskSize,
+  numberOfWorkers: defaultNumDataprocWorkers,
+  numberOfPreemptibleWorkers: defaultNumDataprocPreemptibleWorkers,
+  workerMachineType: defaultDataprocMachineType,
+  workerDiskSize: defaultDataprocWorkerDiskSize,
+  componentGatewayEnabled: true,
+  gpuEnabled: false,
+  hasGpu: false,
+  gpuType: defaultGpuType,
+  numGpus: defaultNumGpus,
+  autopauseThreshold: defaultAutopauseThreshold,
+  computeRegion: defaultComputeRegion,
+  computeZone: defaultComputeZone,
+}
+
+const updateComputeConfig = jest.fn()
+
+const defaultPersistentDiskProps = {
+  diskExists: true,
+  computeConfig: defaultIComputeConfig,
+  updateComputeConfig,
+  handleLearnMoreAboutPersistentDisk: jest.fn(),
+}
+
+const defaultPersistentDiskTypeProps = {
+  diskExists: true,
+  computeConfig: defaultIComputeConfig,
+  updateComputeConfig,
+}
+
+jest.mock('src/libs/notifications', () => ({
+  notify: jest.fn((...args) => {
+    console.debug("######################### notify"); /* eslint-disable-line */
+    console.debug({ method: "notify", args: [...args] }); /* eslint-disable-line */
+  }),
+}))
+
+describe('compute-modal-component', () => {
+  describe('PersistentDiskType', () => {
+    // Passing diskExists [true,false] to PersistentDiskType
+    it('should be disabled when existing PD', async () => {
+      // Act
+      await render(h(PersistentDiskType, defaultPersistentDiskTypeProps))
+      const dType = screen.getByLabelText('Disk Type')
+
+      // Assert
+      expect(dType).toBeDisabled()
+    })
+    it('should not be disabled when no existing PD', async () => {
+      // Act
+      await render(h(PersistentDiskType, { ...defaultPersistentDiskTypeProps, diskExists: false }))
+
+      // Assert
+      const dType = screen.getByLabelText('Disk Type')
+
+      expect(dType).not.toBeDisabled()
+    })
+
+    // Ensuring updateComputeConfig gets called with proper value on change
+    it('should call updateComputeConfig with proper value on change', async () => {
+      // Act
+      render(h(PersistentDiskType, { ...defaultPersistentDiskTypeProps, diskExists: false }))
```

Comment:
I believe `render` here should be in the "arrange" section, since it's setting up for the main point of this test, which is changing the value. Similar comments in other tests: try highlighting the main point of each test in the "act" section.